src/events/channels.service.ts

```diff
@@ -95,6 +104,50 @@ export class ChannelsEventService {
     };
   }
 
+  async usersAdded(
+    client: any,
+    payload: UsersAddedDto,
+  ): Promise<EventReplyDto> {
+    const result = await this.workspaceChannelService.addUsersToChannel(
+      client.user,
+      payload.broadcast.workspace_id,
+      payload.broadcast.channel_id,
+      payload.data,
+    );
+
+    if (!result) {
+      return {
+        status: 'FAILED',
+        error: {
+          id: '500',
+          message: 'Internal Server Error',
+        },
+        seq_reply: payload.seq,
+      };
+    }
+
+    if (
+      !(
+        result instanceof NotFoundException ||
+        result instanceof ForbiddenException
+      )
+    ) {
+      const { addedUsers } = result;
+
+      if (addedUsers.length > 0) {
+        client
+          .to(RoomType.Channel + payload.broadcast.channel_id)
+          .emit(payload.event, addedUsers);
+      }
+    }
```

Comment:
It would be better to distinguish between different error types and provide more specific error messages for better debugging and error reporting.